.circleci/config.yml

```diff
@@ -24,27 +45,63 @@ jobs:
       - run:
           name: Build Docker image
           command: |
-            docker build -t ${IMAGE} -f Dockerfile .
+            docker build -t ${IMAGE}:${CIRCLE_BUILD_NUM} -f Dockerfile .
 
       - run:
           name: Tag images and push images
           command: |
-            docker tag ${IMAGE}:latest ${IMAGE}:${VERSION}-${CIRCLE_BUILD_NUM}
-            docker push ${IMAGE}:${VERSION}-${CIRCLE_BUILD_NUM}
-            docker push ${IMAGE}:latest
+            docker tag ${IMAGE}:${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:${CIRCLE_BRANCH}-${CIRCLE_SHA1:0:7}-${TIMESTAMP}
+            docker tag ${IMAGE}:${CIRCLE_BUILD_NUM} ${DOCKERHUB_IMAGE}:${VERSION}-${CIRCLE_BUILD_NUM}
+            docker tag ${IMAGE}:${CIRCLE_BUILD_NUM} ${DOCKERHUB_IMAGE}:latest
+            docker tag ${IMAGE}:${CIRCLE_BUILD_NUM} ${GCP_IMAGE}:latest
+
+            if [[ ${CIRCLE_BRANCH} == "develop"   ]]; then
+              docker push ${GCP_IMAGE}:${CIRCLE_BRANCH}-${CIRCLE_SHA1:0:7}-${TIMESTAMP}
+              docker push ${GCP_IMAGE}:latest
+            fi
+            if [[ ${CIRCLE_BRANCH} == "master" ]]; then
+              docker push ${DOCKERHUB_IMAGE}:${VERSION}-${CIRCLE_BUILD_NUM}
+              docker push ${DOCKERHUB_IMAGE}:latest
+              docker push ${GCP_IMAGE}:${CIRCLE_BRANCH}-${CIRCLE_SHA1:0:7}-${TIMESTAMP}
```

Comment:
I don't think we need to push to the private registry for master but, if that's the case, you can add an `or` in the first condition and remove it from the master branch. This is to avoid code duplication